plugins/native/module/vtkF3DSPZReader.cxx

```diff
@@ -280,7 +317,7 @@ int vtkF3DSPZReader::RequestData(
   const Header* header = reinterpret_cast<Header*>(uncompressed.data());
 
   assert(header->magic == 0x5053474e);
-  assert(header->version == 2);
+  assert(2 >= header->version <= 3);
```

Comment:
I'm surprised it's legal. Are you sure it's doing what's you have in mind?
To me `2 >= header->version` is evaluated to `true`, then `true <= 3` is evaluated to true again so it works by accident